src/osd/ECUtil.cc

```diff
@@ -71,28 +68,55 @@ int ECUtil::decode(
     need.insert(i->first);
   }
 
-  for (uint64_t i = 0; i < total_data_size; i += sinfo.get_chunk_size()) {
+  set<int> avail;
+  for (auto i = to_decode.begin();
+       i != to_decode.end();
+       ++i) {
+    assert(i->second.length() != 0);
+    avail.insert(i->first);
+  }
+
+  map<int, vector<pair<int, int>>> min;
+  assert(ec_impl->minimum_to_decode(need, avail, &min) == 0);
```

Comment:
@mynaramana we should not rely on the side effect in `assert()`, in future, `assert()` in Ceph could be optimized out if `NDEBUG` is defined.